src/tools/inner_node.rs

```diff
@@ -122,6 +122,12 @@ impl InnerNode {
         Ok(())
     }
 
+    /// Connects to the target address.
+    pub async fn connect_from(&self, target: SocketAddr, socket: TcpSocket) -> io::Result<()> {
+        self.node.connect_using_socket(target, socket).await?;
+        Ok(())
+    }
```

Comment:
```suggestion
    pub async fn connect_from(&self, target: SocketAddr, socket: TcpSocket) -> io::Result<()> {
        self.node.connect_using_socket(target, socket).await
    }
```

Reply:
I'll change it. I've added it the same way that was connect() implemented (to be consistent with other code in inner_node.rs).

Reply:
Aha, okay. Feel free to apply this improvement to that function as well (in some future PR/commit)